### PR TITLE
chore: update rust version + disable unsupported optimizations

### PR DESCRIPTION
**Pull request by Saschl**
Fixes #[issue_no]

## Summary of Changes

~Requires https://github.com/flybywiresim/dev-env/pull/11 and https://github.com/flybywiresim/msfs-rs/pull/17 to be merged first. Docker image also needs to be updated after dev-env change has been merged.~ Done.

Use recent rust version and disable sign extension that is enabled by default and added by wasm-opt, but is unsupported by the msfs runtime. Also fixes some clippy errors that came up with newer versions

## Additional context

Discord username (if different from GitHub):

## Testing instructions

No changes are expected. The plane should behave as usual. 

Especially watch that the electrical and hydraulic system are working as usual.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by Gurgel100** [comment]
You'll need to update the msfs crate once flybywiresim/msfs-rs#17 is merged.

File: fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs [hunk 1]
```diff
@@ -3075,7 +3070,7 @@ impl A320BlueElectricPumpController {
     }
 
     fn has_overheat_fault(&self) -> bool {
-        self.has_low_level_fault
+        self.has_overheat_fault
```
Comment (Saschl) [inline fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:3073]: I hope this is correct, clippy complained and it made sense to me.

Comment (Gurgel100) [reply fbw-a32nx/src/wasm/systems/a320_systems/src/hydraulic/mod.rs:3073]: Looks good also from my side.

**Review by Gurgel100 — APPROVED** [review]

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing:  #8242
Tier of Testing : 1
Date : 02/11/2023

Testing Process:
- Test various systems, perform a flight.

Negatives: 
- Terrain ND overlay shows up in some ND views but its completely borked as shown below:
image
image
image

Notes:
Simbridge was connected before flight started (simbridge running w/ autoconnect on) - no artifacts as the above in latest dev branch. 

Testing Results:
Not Passed

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing: https://github.com/flybywiresim/aircraft/pull/8242
Tier of Testing : 1
Date : 02/11/2023

Testing Process:


- Test various systems, perform a flight.

Testing Results:
Passed

**Pull request merged**